From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrico=20Tr=C3=B6ger?= <enrico.troeger@uvena.de>
Date: Sun, 3 Apr 2022 14:36:36 +0200
Subject: [PATCH] GitChangeBar: Support libgit2 1.4.x

libgit2 1.4.0 changed the public API of "git_buf" and renamed the
attribute "asize" to "reserved".
For details, see https://github.com/libgit2/libgit2/pull/6078/files.

Upstream-Status: Pending
Reference: https://github.com/geany/geany-plugins/pull/1165
---
 git-changebar/src/gcb-plugin.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/git-changebar/src/gcb-plugin.c b/git-changebar/src/gcb-plugin.c
index f8ce20c..43ec77e 100644
--- a/git-changebar/src/gcb-plugin.c
+++ b/git-changebar/src/gcb-plugin.c
@@ -49,6 +49,11 @@
 # define git_buf_dispose  git_buf_free
 # define git_error_last   giterr_last
 #endif
+#if ! defined (LIBGIT2_VER_MINOR) || (LIBGIT2_VER_MAJOR < 1) || ( (LIBGIT2_VER_MAJOR == 1) && (LIBGIT2_VER_MINOR < 4) )
+# define gcb_git_buf_reserved asize
+#else
+# define gcb_git_buf_reserved reserved
+#endif
 
 
 GeanyPlugin      *geany_plugin;
@@ -216,7 +221,7 @@ static int
 gcb_git_buf_grow (git_buf  *buf,
                   size_t    target_size)
 {
-  if (buf->asize == 0) {
+  if (buf->gcb_git_buf_reserved == 0) {
     if (target_size == 0) {
       target_size = buf->size;
     }
@@ -234,7 +239,7 @@ buf_zero (git_buf *buf)
   if (buf) {
     buf->ptr = NULL;
     buf->size = 0;
-    buf->asize = 0;
+    buf->gcb_git_buf_reserved = 0;
   }
 }
 
